synapseclient/api/configuration_services.py

```diff
@@ -0,0 +1,136 @@
+"""This module is responsible for exposing access to any configuration either through
+file, environment variables, or other means.
+"""
+
+import configparser
+import functools
+from typing import Dict
+
+from synapseclient.core.constants import config_file_constants
+from synapseclient.core.pool_provider import DEFAULT_NUM_THREADS
+
+
+@functools.lru_cache()
+def get_config_file(config_path: str) -> configparser.RawConfigParser:
+    """
+    Retrieves the client configuration information.
+
+    Arguments:
+        config_path:  Path to configuration file on local file system
+
+    Returns:
+        A RawConfigParser populated with properties from the user's configuration file.
+    """
+
+    try:
+        config = configparser.RawConfigParser()
+        config.read(config_path)  # Does not fail if the file does not exist
```

Comment:
Do we want to add error-handling to address the comment here?

Reply:
No. I think leaving this as is is the appropriate choice. The reason is that there are multiple authentication methods and the config file is only one of them. In the case the file is not found it should fall through to the next auth method. Then only at the end once no other methods can authenticate you are we raising an exception.